python/lsst/pipe/tasks/script/registerDcrSubfilters.py

```diff
@@ -0,0 +1,60 @@
+# This file is part of pipe_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import click
+from collections import defaultdict
+import logging
+from sqlalchemy.exc import IntegrityError
+
+from lsst.daf.butler import Butler
+
+
+_log = logging.getLogger(__name__.partition(".")[2])
+
+
+def registerDcrSubfilters(repo, num_subfilters, filter_names):
+    """Construct a set of subfilters for chromatic modeling and add them to a
+    registry.
+
+    Parameters
+    ----------
+    repo : `str`
+        URI to the location to read the repo.
+    num_subfilters : `int`
+        The number of subfilters to add.
+    filter_names : `list` [`str`]
+        The filter names to add.
+    """
+    butler = Butler(repo, writeable=True)
+    records = defaultdict(list)
+    with butler.registry.transaction():
+        record = []
+        for filterName in filter_names:
+            for sub in range(num_subfilters):
+                record.append({"band": filterName, "subfilter": sub})
+                records[filterName].append(sub)
```

Comment:
I don't actually see that we're using `records` anywhere below.

And if we can just get rid of it, maybe we can rename that `record` list to `records`?

Reply:
yes, it's a remnant from trying a different way to generate the result string. Thanks for catching, will remove.